### PR TITLE
Don't count pending requests towards underprovisioning

### PR DESCRIPTION
**Pull request by pschoenfelder**

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/data/StateManager.java [hunk 1]
```diff
@@ -195,6 +198,7 @@ public SingularityState generateState(boolean includeRequestIds) {
       updatePossiblyUnderProvisionedAndOverProvisionedIds(requestWithState, numInstances, overProvisionedRequestIds, possiblyUnderProvisionedRequestIds);
     }
 
+    removePendingRequestIds(possiblyUnderProvisionedRequestIds);
```
Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/data/StateManager.java]: can we give this a better name? Like `filterForPendingRequests` or something more descriptive like that?

File: SingularityService/src/test/java/com/hubspot/singularity/data/StateManagerTest.java [hunk 2]
```diff
@@ -53,4 +57,21 @@ public void itDoesntCountCleaningTasks() {
...
   }
+
+  @Test
+  public void itDoesntFlagPendingRequestsForUnderOrOverProvisioning() {
+    initRequest();
+    initFirstDeploy();
+
+    SingularityRequest request = requestResource.getRequest(requestId, singularityUser).getRequest();
+
+    requestManager.activate(request.toBuilder().setInstances(Optional.of(0)).build(), RequestHistoryType.UPDATED, System.currentTimeMillis(), Optional.<String> absent(), Optional.<String> absent());
```
Comment (ssalinas) [inline SingularityService/src/test/java/com/hubspot/singularity/data/StateManagerTest.java]: did you confirm that this test actually fails without the filter you added? Seems odd to have it at instances of `0` (tbh I thought our validation would catch that...)

**Comment by pschoenfelder** [comment]
Yeah, I commented out my original code, wrote the failing test, then
uncommented the code and it passed.

**Comment by ssalinas** [comment]
Cool, let's move this one through the environments and get it merged. 🚢

**Pull request merged**